internal/middleware/logging.go

```diff
@@ -2,37 +2,69 @@ package middleware
 
 import (
 	"context"
+	"encoding/json"
 	"time"
 
 	"github.com/openfga/openfga/pkg/logger"
 	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
 )
 
+const (
+	methodKey       = "method"
+	requestIDKey    = "request_id"
+	traceIDKey      = "trace_id"
+	reqDurationKey  = "req_duration_ms"
+	rawRequestKey   = "raw_request"
+	rawResponseKey  = "raw_response"
+	publicErrorKey  = "public_error"
+	grpcErrorKey    = "grpc_error"
+	grpcCompleteKey = "grpc_complete"
+)
+
 func NewLoggingInterceptor(logger logger.Logger) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		start := time.Now()
 
-		resp, err := handler(ctx, req)
+		fields := []zap.Field{zap.String(methodKey, info.FullMethod)}
+
+		if requestID, ok := RequestIDFromContext(ctx); ok {
+			fields = append(fields, zap.String(requestIDKey, requestID))
+		}
 
-		fields := []zap.Field{
-			zap.String("method", info.FullMethod),
-			zap.Duration("took", time.Since(start)),
+		spanCtx := trace.SpanContextFromContext(ctx)
+		if spanCtx.HasTraceID() {
+			fields = append(fields, zap.String(traceIDKey, spanCtx.TraceID().String()))
 		}
 
+		jsonReq, err := json.Marshal(req)
+		if err == nil {
+			fields = append(fields, zap.Any(rawRequestKey, json.RawMessage(jsonReq)))
+		}
+
+		resp, err := handler(ctx, req)
+
+		fields = append(fields, zap.Duration(reqDurationKey, time.Since(start)))
+
 		if err != nil {
 			if internalError, ok := err.(serverErrors.InternalError); ok {
 				fields = append(fields, zap.Error(internalError.Internal()))
 			}
 
-			fields = append(fields, zap.String("public_error", err.Error()))
-			logger.Error("grpc_error", fields...)
+			fields = append(fields, zap.String(publicErrorKey, err.Error()))
+			logger.Error(grpcErrorKey, fields...)
 
 			return nil, err
 		}
 
-		logger.Info("grpc_complete", fields...)
+		jsonResp, err := json.Marshal(resp)
+		if err == nil {
+			fields = append(fields, zap.Any(rawResponseKey, json.RawMessage(jsonResp)))
+		}
+
+		logger.Info(grpcCompleteKey, fields...)
```

Comment:
Debug level. This is too noisy for Info level logs.

Reply:
I'm torn. If we do that then we will only have error logs. Which may be what we want? 🤷 Let me create a poll just to get a sense of things.